### PR TITLE
[charts] Add missing `themeAugmentation` in pro plan

### LINKED ISSUE #14304: [charts] Missing `themeAugmentation` in pro plan
The `themeAugmentation` is missing in the pro plan of the charts. It would be nice if the pro plan exports its own `themeAugmentation` that extends the community plan and implements the augmentation for the pro components. If that's okay, I could implement that too.

**Search keywords**:

### PR DESCRIPTION
**Pull request by lhilgert9**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Fixes #14304.

Preview: https://deploy-preview-14313--material-ui-x.netlify.app/x/react-charts/getting-started/#typescript

### CODE REVIEW

**Review by JCQuintas — APPROVED** [review]
Thanks for the improvement. 💪 

This looks good to me, I'll just ask my colleague @alexfauquette for a final check. :)

**Comment by alexfauquette** [comment]
Nice improvement. There are just two points we need to verify before merging:

- Is the theme augmentation effectively working?
- Since this PR impacts a non MIT library, we light need you to sign an agreement. I'm not sure about the exact process.' @michelengelen do you have more information

**Comment by michelengelen** [comment]
> Nice improvement. There are just two points we need to verify before merging:
> 
> 
> 
> - Is the theme augmentation effectively working?
> 
> - Since this PR impacts a non MIT library, we light need you to sign an agreement. I'm not sure about the exact process.' @michelengelen do you have more information

@joserodolfofreitas would you mind helping out with the CLA?

**Comment by lhilgert9** [comment]
> > Nice improvement. There are just two points we need to verify before merging:
> > 
> > * Is the theme augmentation effectively working?
> > * Since this PR impacts a non MIT library, we light need you to sign an agreement. I'm not sure about the exact process.' @michelengelen do you have more information
> 
> @joserodolfofreitas would you mind helping out with the CLA?

@michelengelen I must have signed the CLA last year or so, if it is still valid.

**Comment by lhilgert9** [comment]
> * Is the theme augmentation effectively working?
@alexfauquette Should I create an own test file for themeAugmentation or test it in the heatmap test files or how did you imagine that?

**Review by oliviertassinari — COMMENTED** [review]
> Since this PR impacts a non MIT library, we light need you to sign an agreement. I'm not sure about the exact process.'

We document the process for maintainers at the same place as for contributors: https://www.notion.so/mui-org/CLA-Contributor-License-Agreement-92ece655b1584b10b00e4de9e67eedb0?pvs=4#417626432f9e419f9fc8bc90aa6ca871.

> I must have signed the CLA last year or so, if it is still valid.

@lhilgert9 We have an entry for it, but couldn't verify your email. Could you do this step? https://www.notion.so/mui-org/CLA-Contributor-License-Agreement-92ece655b1584b10b00e4de9e67eedb0?pvs=4#fb0396ecff384a4cb515ab9f4351eddc Thanks

**Comment by lhilgert9** [comment]
@oliviertassinari Done👍🏼

**Comment by oliviertassinari** [comment]
@lhilgert9 Ok, thanks, that should be enough in this case, we can clearly correlate the signature form timestamp and link with the GitHub profile.

**Comment by alexfauquette** [comment]
> Is that intentional

Not really. It's because we are in the first version of the library, and did not find time to figure out which component should have this option. We have #12070 to track it

**Comment by alexfauquette** [comment]
@lhilgert9 I'm working on fixing it.

I propose to add the default props to all the single components (LineChart, BarChart, ... LineChartPro, ..., Heatmap)
Not to the subcomponents (LinePlot, LineElement, ..., HeatmapPlot, HeatmapItem) because those components props are just SVG attributes, then modifying them is either for style customization (this can be done with CSS) or custom logic that I would not recommend doing with `defaultProps`

While working on it, I'm cleaning the TS to remove type definition of stuff that don't work

**Comment by lhilgert9** [comment]
@alexfauquette If I can help anywhere, please feel free to let me know.

**Comment by alexfauquette** [comment]
I did a big chunk of cleaning the TS definition.

I propose we merge this PR to get the big picture merged, and in follow-up PR add conformance test, to make sure each coponent have the default Props and styling applied as expected.

**Comment by lhilgert9** [comment]
@alexfauquette The plan sounds good to me. However, I would implement the useThemeProps hook in the code as follows to be in compliance with the other packages:
```ts
function Component(inProps: ComponentProps) {
const props = useThemeProps(inProps);
// implementation
}
```
The rest of the changes LGTM.

File: packages/x-charts/src/internals/components/AxisSharedComponents.tsx
Comment (oliviertassinari) [inline packages/x-charts/src/internals/components/AxisSharedComponents.tsx:18]: I'm confused about this change. Does ``[`.${axisClasses.line}`]: {`` target the same element or a child element? I always find this API confusing. I think we should always have `&` so the developers looking at  the source to see what to customize with CSS modules never have doubts.

Comment (alexfauquette) [reply packages/x-charts/src/internals/components/AxisSharedComponents.tsx:18]: My issue is that we do not `styled()` the sub part of the axis, to avoid styling to many elements.

The idea of this modification was to reduce the specificity of the default style to let people override it from the root with `& .${axisClasses.line}`

Comment (oliviertassinari) [reply packages/x-charts/src/internals/components/AxisSharedComponents.tsx:18]: Does it mean its equivalent to?

```suggestion
[`&.${axisClasses.line}`]: {
```

Comment (alexfauquette) [reply packages/x-charts/src/internals/components/AxisSharedComponents.tsx:18]: I got confused. I thaugh that removing the `&` would remove the root class. But it does not.

This modification changes nothing. I just tried the before after:


```js
[`& .${axisClasses.line}`]: { ... },
[`.${axisClasses.line}`]: { },
```

The both selector above result into the same CSS

```css
.css-6ic30x-MuiChartsAxis-root .MuiChartsAxis-line { ... } // in dev env
.css-6ic30x .MuiChartsAxis-line { ... } // in prod env
```

Comment (alexfauquette) [reply packages/x-charts/src/internals/components/AxisSharedComponents.tsx:18]: Fix in https://github.com/mui/mui-x/pull/14378

**Pull request merged**
